google/resource_container_cluster.go

```diff
@@ -709,41 +709,17 @@ func resourceContainerClusterRead(d *schema.ResourceData, meta interface{}) erro
 	}
 
 	d.Set("name", cluster.Name)
-
 	d.Set("network_policy", flattenNetworkPolicy(cluster.NetworkPolicy))
-
 	d.Set("zone", cluster.Zone)
 
 	locations := schema.NewSet(schema.HashString, convertStringArrToInterface(cluster.Locations))
 	locations.Remove(cluster.Zone) // Remove the original zone since we only store additional zones
 	d.Set("additional_zones", locations)
 
 	d.Set("endpoint", cluster.Endpoint)
-
-	if cluster.MaintenancePolicy != nil {
-		d.Set("maintenance_policy", flattenMaintenancePolicy(cluster.MaintenancePolicy))
-	}
-
-	masterAuth := []map[string]interface{}{
-		{
-			"username":               cluster.MasterAuth.Username,
-			"password":               cluster.MasterAuth.Password,
-			"client_certificate":     cluster.MasterAuth.ClientCertificate,
-			"client_key":             cluster.MasterAuth.ClientKey,
-			"cluster_ca_certificate": cluster.MasterAuth.ClusterCaCertificate,
-		},
-	}
-	if len(cluster.MasterAuth.ClientCertificate) == 0 {
-		masterAuth[0]["client_certificate_config"] = []map[string]interface{}{
-			{"issue_client_certificate": false},
-		}
-	}
-	d.Set("master_auth", masterAuth)
-
-	if cluster.MasterAuthorizedNetworksConfig != nil {
-		d.Set("master_authorized_networks_config", flattenMasterAuthorizedNetworksConfig(cluster.MasterAuthorizedNetworksConfig))
-	}
-
+	d.Set("maintenance_policy", flattenMaintenancePolicy(cluster.MaintenancePolicy))
+	d.Set("master_auth", flattenMasterAuth(cluster.MasterAuth))
+	d.Set("master_authorized_networks_config", flattenMasterAuthorizedNetworksConfig(cluster.MasterAuthorizedNetworksConfig))
```

Comment:
Could we check the error on these? Complex types with `d.Set` can sometimes get tricky. :/